src/bin/activate.rs

```diff
@@ -273,16 +278,16 @@ pub async fn activation_confirmation(
 
     watcher.watch(&lock_path, RecursiveMode::NonRecursive)?;
 
-    if let Err(err) = danger_zone(done, confirm_timeout).await {
-        error!("Error waiting for confirmation event: {}", err);
+    // if let Err(err) = danger_zone(done, confirm_timeout).await {
```

Comment:
I had to disable this to get it to work on my machine, out of luck I managed to get this:
```
🚀 ℹ️ [deploy] [INFO] The following profiles are going to be deployed:
[blowhole.system]
user = "root"
ssh_user = "main"
path = "/nix/store/h33hp9r4br258fkry6fs9hqrbimillrz-activatable-nixos-system-blowhole-22.05.20211206.bc5d683"
hostname = "blowhole"
ssh_opts = []

🚀 ❓ [deploy] [DEBUG] Finding the deriver of store path for /nix/store/h33hp9r4br258fkry6fs9hqrbimillrz-activatable-nixos-system-blowhole-22.05.20211206.bc5d683
🚀 ℹ️ [deploy] [INFO] Building profile `system` for node `blowhole`
warning: unknown experimental feature 'ca-references'
🚀 ℹ️ [deploy] [INFO] Copying profile `system` to node `blowhole`
warning: unknown experimental feature 'ca-references'
🚀 ℹ️ [deploy] [INFO] Activating profile `system` for node `blowhole`
🚀 ❓ [deploy] [DEBUG] Constructed activation command: sudo -u root /nix/store/h33hp9r4br258fkry6fs9hqrbimillrz-activatable-nixos-system-blowhole-22.05.20211206.bc5d683/activate-rs --debug-logs activate '/nix/store/h33hp9r4br258fkry6fs9hqrbimillrz-activatable-nixos-system-blowhole-22.05.20211206.bc5d683' '/nix/var/nix/profiles/system' --temp-path '/tmp' --confirm-timeout 30 --magic-rollback --auto-rollback
🚀 ❓ [deploy] [DEBUG] Constructed wait command: /nix/store/h33hp9r4br258fkry6fs9hqrbimillrz-activatable-nixos-system-blowhole-22.05.20211206.bc5d683/activate-rs --debug-logs wait '/nix/store/h33hp9r4br258fkry6fs9hqrbimillrz-activatable-nixos-system-blowhole-22.05.20211206.bc5d683' --temp-path '/tmp'
🚀 ℹ️ [deploy] [INFO] Creating activation waiter
[sudo] password for main: 
Sorry, try again.
[sudo] password for main: 
⭐ ℹ️ [activate] [INFO] Activating profile
⭐ ❓ [activate] [DEBUG] Running activation script
updating GRUB 2 menu...
activating the configuration...
setting up /etc...
reloading user units for main...
setting up tmpfiles
⭐ ℹ️ [activate] [INFO] Activation succeeded!
⭐ ℹ️ [activate] [INFO] Magic rollback is enabled, setting up confirmation hook...
⭐ ❓ [activate] [DEBUG] Ensuring parent directory exists for canary file
⭐ ❓ [activate] [DEBUG] Creating canary file
⭐ ❓ [activate] [DEBUG] /tmp/deploy-rs-canary-h33hp9r4br258fkry6fs9hqrbimillrz
⭐ ❓ [activate] [DEBUG] Creating notify watcher
⭐ ℹ️ [activate] [INFO] Waiting for confirmation event...
🚀 ❓ [deploy] [DEBUG] Wait command ended Ok(Output { status: ExitStatus(unix_wait_status(0)), stdout: "", stderr: "👀 ℹ️\u{fe0f} [wait] [\u{1b}[38;5;51mINFO\u{1b}[0m] Waiting for confirmation event...\n👀 ℹ️\u{fe0f} [wait] [\u{1b}[38;5;51mINFO\u{1b}[0m] Found canary file, done waiting!\n" })
                                                             🚀 ℹ️ [deploy] [INFO] Success activating, attempting to confirm activation
                   🚀 ❓ [deploy] [DEBUG] Attempting to run command to confirm deployment: rm /tmp/deploy-rs-canary-h33hp9r4br258fkry6fs9hqrbimillrz
                                 🚀 ℹ️ [deploy] [INFO] Deployment confirmed.
                                                                           ⭐ ❌ [activate] [ERROR] Error waiting for confirmation event: Timeout elapsed for confirmation
⭐ ⚠️ [activate] [WARN] De-activating due to error
switching profile from version 7 to 6
⭐ ❓ [activate] [DEBUG] Listing generations
⭐ ❓ [activate] [DEBUG] Removing generation entry    7   2021-12-08 23:29:07   
⭐ ⚠️ [activate] [WARN] Removing generation by ID 7
removing profile version 7
⭐ ℹ️ [activate] [INFO] Attempting to re-activate the last generation
updating GRUB 2 menu...
activating the configuration...
setting up /etc...
reloading user units for main...
setting up tmpfiles
⭐ ❓ [activate] [DEBUG] Got worthy removal event, sending on channel
⭐ ❌ [activate] [ERROR] Could not send file system event to watcher: channel closed
```
notice the inotify event arriving only at the absolute end, I couldn't reproduce it again, no clue what's happening here